hare3/graded_gossip.go

```diff
@@ -0,0 +1,97 @@
+package hare3
+
+import "github.com/spacemeshos/go-spacemesh/common/types"
+
+// GradedGossiper works as a filter, for the given message inputs it returns
+// one of 3 results indicating what action to take with the message it was
+// invoked for.
+type GradedGossiper interface {
```

Comment:
what the point of it being an interface? does it simplify testing?

Reply:
Same answer as above